src/main/kotlin/org/elm/workspace/ElmProject.kt

```diff
@@ -333,6 +356,19 @@ private class ElmApplicationProjectDTO(
 ) : ElmProjectDTO
 
 
+/**
+ * DTO used to wrap the data in `elm.intellij.json`.
+ *
+ * @see [ElmToolchain.ELM_INTELLIJ_JSON]
+ */
+private class ElmCustomManifestDTO(
+        /**
+         * The path to the directory containing the unit tests, relative to the root of the Elm project.
+         */
+        @JsonProperty("test-directory") val testDirectory: String?
```

Comment:
Why make this String nullable?

Reply:
Yeah, good point. I was thinking that in future we might add more to this file, and the user might want to set the other fields that we might add, but not to set this one, so the DTO should allow this to be null.  But then... YAGNI 😄 .  Will make it non-nullable!